x/bridge/keeper/transfers.go

```diff
@@ -31,90 +26,96 @@ func (k Keeper) InboundTransfer(
 	params := k.GetParams(ctx)
 
 	// Check if the asset accepts inbound transfers
-	asset, ok := params.GetAsset(assetID)
-	if !ok {
+	asset, found := params.GetAsset(assetID)
+	if !found {
 		return errorsmod.Wrapf(types.ErrInvalidAssetID, "Asset not found %s", assetID.Name())
 	}
 	if !asset.Status.InboundActive() {
 		return errorsmod.Wrapf(types.ErrInvalidAssetStatus, "Inbound transfers are disabled for this asset")
 	}
 
-	// Try to finalize the transfer in store
-	err := k.finalizeInboundTransfer(ctx, externalID, sender, destAddr, assetID, amount, params.VotesNeeded)
-	switch {
-	case err == nil:
-		// Everything is fine, mint!
-	case errors.Is(err, ErrAlreadyFinalized), errors.Is(err, ErrNeedMoreVotes):
-		// Expected scenario, just return without minting
-		return nil
-	default:
-		// Unexpected error
+	// Try to finalize the transfer
+	finalized, err := k.finalizeInboundTransfer(
+		ctx,
+		externalID,
+		externalHeight,
+		sender,
+		destAddr,
+		assetID,
+		amount,
+		params.VotesNeeded,
+	)
+	if err != nil {
 		return errorsmod.Wrapf(sdkerrors.ErrLogic, "Can't finalize inbound trander: %s", err.Error())
 	}
+	if !finalized {
+		// The transfer either doesn't have enough votes or has already been finalised
+		return nil
+	}
```

Comment:
I think the verbiage might be a little confusing, can we call it "finalizeOrVoteOnInboundTransfer", since we are doing two things here? Or maybe we even split this up where we first vote on the transfer, and then call a secondary of attempting to finalize it based on percent of votes

Reply:
I don't like separation, I think it would be inconvenient and not really clear since we will probably need to get/update the transfer twice. What do you think of "voteAndFinalize" naming? For me it's quite staightforward

Reply:
voteAndFinzalize sounds good to me